jobs/openshift4/cluster/integreatly/install/Jenkinsfile

```diff
@@ -0,0 +1,94 @@
+
+String cluserAPI = "https://api.${CLUSTER_DOMAIN}:6443"
+Boolean cleanup = params.CLEANUP
+
+
+node('cirhos_rhel7') {
+     timeout(60) { ansiColor('gnome-terminal') { timestamps {
+    //currentBuild.displayName = "${currentBuild.displayName} ${CLUSTER_DOMAIN}"
+    currentBuild.description = "cluster: ${CLUSTER_DOMAIN}\n"
+
+
+    stage('Login to OpenShift 4 cluster') {
+        sh "oc login ${cluserAPI} --insecure-skip-tls-verify=true -u ${ADMIN_USERNAME} -p ${ADMIN_PASSWORD}"
+    } 
+
+    stage('Clone Integreatly Operator') {
+        dir('integreatly-operator') {
+            checkout([
+                $class: 'GitSCM',
+                branches: [[name: "${INTEGREATLY_OPERATOR_BRANCH}"]],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [],
+                userRemoteConfigs: [[url: "${INTEGREATLY_OPERATOR_REPOSITORY}"]]
+            ])
+            sh "git clean -xdf"
+        } 
+    }
+
+    stage('Clean up') {
+        when (params.CLEANUP){
+            println "Cleaning..."
+            dir("integreatly-operator") {
+                sh "make cluster/cleanup/olm"
+                sh "make cluster/cleanup/crds" 
+            }
+        }
+    }
+
+    stage('Create Integreatly Operator') {
+        sh "export AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}"
+        sh "export AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}"
+        sh "export AWS_BUCKET=${AWS_BUCKET}"
+        sh "export GH_CLIENT_ID=${GH_CLIENT_ID}"
+        sh "export GH_CLIENT_SECRET=${GH_CLIENT_SECRET}"
+
+        dir("integreatly-operator") {
+            sh "make cluster/prepare/olm"  
+        }
+    }
+
+    stage ('Create Installation CR') {
+        dir("integreatly-operator") {
+            sh "make deploy/integreatly-installation-cr.yml"
+            sh "oc create -f deploy/integreatly-installation-cr.yml"   
+        }
+    }
+
+    stage ('Verify Installation Has Finished') {
+        String getAuthStatusCommand = "oc get Installation example-installation -n integreatly --output=json -o jsonpath='{.status.stages.authentication.phase}'"
+        expectedOcCommandOutput = getInstallationStatus(getAuthStatusCommand)
+        while (!expectedOcCommandOutput.contains("completed")) {
+            println "Authentication stage has not completed yet"
+            sleep time: 20, unit: 'SECONDS'
+            expectedOcCommandOutput = getInstallationStatus(getAuthStatusCommand)
+        }  
+
+        String getInstallationStatusCommand = "oc get Installation example-installation -n integreatly --output=json -o jsonpath='{.status.stages.products.phase}'"
+        expectedOcCommandOutput = getInstallationStatus(getInstallationStatusCommand)
+        while (!expectedOcCommandOutput.contains("completed")) {
+            println "Installation has not completed yet"
+            sleep time: 20, unit: 'SECONDS'
+            expectedOcCommandOutput = getInstallationStatus(getInstallationStatusCommand)
+        }
+    }
```

Comment:
We should probably consider making the make file do the wait so we can make it consistent across all installations (jenkins vs prow vs local) https://github.com/integr8ly/integreatly-operator/blob/master/Makefile#L116-L118

We could separate that out into a different target that only does the deploy

Reply:
Leave as it is for now